common/ptr.h

```diff
@@ -285,6 +228,11 @@ class SharedPtr : public BasePtr<T> {
 	SharedPtr(std::nullptr_t) : BasePtr<T>() {
 	}
 
+	~SharedPtr() {
+		if (this->_tracker)
```

Comment:
Unfortunately it's actually not superfluous because _tracker is in an argument-dependent base class.  It will fail to compile without it.  The alternative would be to just remove the base class and copy _tracker into the individual classes (which might be better anyway since it's not doing much good now).